openassessment/xblock/submission_mixin.py

```diff
@@ -250,8 +276,24 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
 
         return {'success': True, 'msg': u''}
 
+    def create_team_submission(self, student_item_dict, student_sub_data):
+        """ A student submitting for a team should generate matching submissions for every member of the team """
+        team_dict = self.get_team_info()
+
+        if team_dict is not None:
+            submissions = []
+
+            team_usernames = team_dict["team_usernames"]
+
+            # submissions are identical except for the student ID
+            for student_id in team_usernames:
```

Comment:
Here you'll want user ids (or maybe anonymized ids?) instead of usernames.  You'll have to modify the `get_team_info()` method to get it.  We can chat about how to do this.